ui/app/src/elements/components/render-block.ts

```diff
@@ -22,7 +22,7 @@ export class RenderBlock extends ScopedElementsMixin(LitElement) {
 
   renderRenderer(element: Element | undefined) {
     if (element) {
-      this.renderer(element as HTMLElement, this.registry);
+      this.renderer(element as HTMLElement, customElements);
```

Comment:
Hrmmmm, is that actually correct? I would have thought this could cause duplicate registration errors since all elements are being registered into the global registry rather than a component-local one. So if two different elements contain (eg) `NHButton` then that will throw an error, and whichever component was instantiated last will fail to render its children.

The problem is more likely to be that `this.registry` has a different name with the different mixin, and we need to access it differently.

Reply:
Ah okay, that makes sense. I wasn't totally sure what was going on tbh and this got it working. But what your are saying is that since we changed the parent class which handles scoped elements differently, we thereby have a different component specific custom element registry?

Does that have to do with shadow DOM stuff?

Reply:
Nope, scratch that. The property is indeed called `registry`. Although it [may have to be](https://github.com/lit/lit/blob/579163218a7b373fdd500872b763d12636c9c5dd/packages/labs/scoped-registry-mixin/src/scoped-registry-mixin.ts#L35C12-L35C20) accessed via `this.constructor.registry`?

In any case, I'll add some tests around this.

Reply:
Oh I see. `this.registry` is a custom getter in our own class? I'm really not sure why that was done, I would presume that the mixin provides it. Also not seeing where it gets assigned yet. But I really doubt that `appletRenderers` needs to determine and provide appropriate registries, since the components should just encapsulate their own. But I could be wrong. Will keep investigating.

Reply:
Oh sorry @weswalla just saw your comment. I just added a WIP `render-block` test that hopefully shows what I'm trying to test. I'm pretty sure once those assertions are working they should fail, since `Child1` and `Child2` will both try to register `NHButton` to the global `customElements` within the `RenderBlock` class. Probably we need to figure out what the new component-specific registry is called and pass that instead.

Maybe you can finish this off? The interface is a bit strange for the uninitiated and I'm not very well, brain is consequently sleepy...

Reply:
Got it, went through a bit of a dive into `MatrixStore` to figure it out but I think what's set up now is the expected API.

Couple more assertions to run. Turns out this passes fine but I think it might actually be *nesting* of `RenderBlocks` that could cause a crash.